src/main/java/data/SpendingAnalytics.java

```diff
@@ -17,33 +17,78 @@
 import com.google.appengine.api.datastore.Entity;
 import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableSet;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.stream.Collectors;
 
 /** Class that computes and stores user's spending analytics. */
 public class SpendingAnalytics {
   private final HashMap<String, Double> storeAnalytics;
+  private final HashMap<String, Double> categoryAnalytics;
 
   public SpendingAnalytics(ImmutableSet<Entity> allReceipts) {
     storeAnalytics = new HashMap<>();
+    categoryAnalytics = new HashMap<>();
 
     for (Entity receipt : allReceipts) {
-      String store = (String) receipt.getProperty("store");
-      Double price = (Double) receipt.getProperty("price");
-
-      // Don't add to hashmap if either store or price are invalid.
-      if (!Strings.isNullOrEmpty(store) && price != null) {
-        if (storeAnalytics.containsKey(store)) {
-          storeAnalytics.put(store, storeAnalytics.get(store) + price);
-        } else {
-          storeAnalytics.put(store, price);
-        }
+      updateStoreAnalytics(receipt);
+
+      updateCategoryAnalytics(receipt);
+    }
+  }
+
+  /** Updates storeAnalytics hashmap with store info from the passed in receipt. */
+  private void updateStoreAnalytics(Entity receipt) {
+    String store = (String) receipt.getProperty("store");
+    Double price = (Double) receipt.getProperty("price");
+
+    // Don't add to hashmap if either store or price are invalid.
+    if (Strings.isNullOrEmpty(store) || price == null) {
+      return;
+    }
+
+    // Either update the existing entry or add a new one if it doesn't exist.
+    if (storeAnalytics.containsKey(store)) {
+      storeAnalytics.put(store, storeAnalytics.get(store) + price);
+    } else {
+      storeAnalytics.put(store, price);
+    }
+  }
+
+  /** Updates categoryAnalytics hashmap with category info from the passed in receipt. */
+  private void updateCategoryAnalytics(Entity receipt) {
+    Double price = (Double) receipt.getProperty("price");
+
+    // Don't add any categories to hashmap if price is invalid.
+    if (price == null) {
+      return;
+    }
+
+    ImmutableSet<String> categories;
+
+    // If categories is null for a particular receipt, just skip it and continue.
+    try {
+      categories = ImmutableSet.copyOf((ArrayList) receipt.getProperty("categories"));
+    } catch (NullPointerException exception) {
```

Comment:
See Item 69 from Effective Java: Use exceptions only for exceptional conditions 

It would be better to check for `receipt.getProperty("categories") == null`